docs/02-guides/maintenance-mode/automatic-detection-with-service-health-checks.mdx

```diff
@@ -0,0 +1,104 @@
+---
+title: Automatic detection with service Health Checks
+description:
+  Learn how to implement health checks to detect down services and automatically
+  place your site in maintenance mode.
+sidebar_position: 2
+---
+
+<SinceVersion tag="3.6" />
+
+# Health Checks
+
+Health checks allow you to monitor the availability of your services and
+automatically put your store in maintenance mode if a service is down. This
+ensures that your users are not affected by a service outage.
+
+With health checks, you can define a set of services that you want to monitor
+and provide a function to check the health of each service. The maintenance mode
+API will periodically run these health check functions and if a service is down,
+it will automatically put the store in maintenance mode. Once the service is
+back up, the maintenance mode API will automatically remove the store from
+maintenance mode.
+
+Additionally, you can configure the scheduling of the health checks using a Cron
+pattern. This allows you to control how often the health checks are run and how
+quickly your store is put in maintenance mode if a service is down.
+
+:::info
+
+When Maintenance Mode has been activated manually via
+[the maintenance mode API](./entering-and-exiting-maintenance-mode#the-maintenance-mode-api),
+the health checks will not remove the store from maintenance mode even after a
+service has recovered. To remove the store from maintenance mode, you must
+[manually deactivate](./entering-and-exiting-maintenance-mode#deactivating-the-maintenance-mode)
+it.
+
+:::
+
+### Adding a Service Health Check
+
+To add a health check, you need to define a function that checks the health of
+your service and returns:
+
+- `true`: when the service is `up`
+- `false`: when the service is `down`
+- `thrown Error`: assumes the service is `down`
+
+This function is then passed to `addHealthCheckService` in the `onServerInit`
+hook in your extension definition.
+
+Here is an example:
+
+```ts title="extensions/health-check.ts"
```

Comment:
**thought:** Shouldn't this path be in an extension, rather than in the extensions' "root" directory?

```suggestion
```ts title="extensions/acme/health-check.ts"
```

Reply:
I do believe we should encourage integrator to organise their extension in separated folders, even if those are 1 file big, but I guess you're right.